app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -2,7 +2,14 @@ module Katello
   class Api::V2::RepositoriesController < Api::V2::ApiController # rubocop:disable Metrics/ClassLength
     include Katello::Concerns::FilteredAutoCompleteSearch
 
-    wrap_parameters :repository, :include => RootRepository.attribute_names.concat([:ignore_global_proxy])
+    generic_repo_wrap_params = []
+    RepositoryTypeManager.generic_remote_options.each do |option|
```

Comment:
I think this needs to use Defined_only: true as well, as the class gets loaded at startup, so this will be executed only once when the rails app loads

Reply:
If you agree, i think this is the last thing!  Otherwise this looks good to me

Reply:
@jlsherrill this seemed to work with testing. what would be case where enabled_only being true might because an issue?

Reply:
@rverdile one way to test this out would be to clear out the pulp primary smart proxy's capabilities, stick a debugger in the code here, and restart your server. I'm pretty sure you'd see the enabled content types list would be empty by the time this is run.  If the smart proxy is corrected with a refresh after that, this class would still be loaded with the incorrect wrap_parameters.